spring-cloud-gcp-vision/src/main/java/com/google/cloud/spring/vision/CloudVisionTemplate.java

```diff
@@ -78,6 +94,31 @@ public String extractTextFromImage(Resource imageResource, ImageContext imageCon
     return result;
   }
 
+  /**
+   * Extract the text out of a file and return the result as a String.
+   *
+   * @param fileResource the file one wishes to analyze
+   * @param mimeType the mime type of the fileResource
+   * @return the text extracted from the pdf aggregated to a String
+   * @throws CloudVisionException if the image could not be read or if text extraction failed
+   */
+  public String extractTextFromFile(Resource fileResource, String mimeType) {
+    AnnotateFileResponse response = analyzeFile(fileResource, mimeType, Type.TEXT_DETECTION);
+
+    List<AnnotateImageResponse> annotateImageResponses = response.getResponsesList();
```

Comment:
The 10 lines of code below are the same as what we need to do for `extractTextFromImage` and should be refactored into a shared private utility method.

Reply:
With the latest changes, the 10 lines below are not the exact same anymore. The return value changed now to a list of strings instead of a single string. Do you think there are still some lines left that are worth moving into a separate utility method?

Reply:
This actually got me thinking, for the sake of simplicity of use, maybe we should just return a single concatenated string, separated by newlines, or maybe a customizable separator that can be configured on the class level. What do you think?

Reply:
In my opinion, this really depends on what the caller is doing with the extracted text. If you want to process the text page by page it does not make too much sense to merge it into one string. But by returning a list of strings you leave the option open to the caller how he then wants to handle the response.

Reply:
I was just wondering if the most common usecase would be to process only 1 page anyway. In that case, a list would make it a bit more verbose to use. Which option seems more convenient for your usecase, for example?